### DISCUSSION TITLE
Add native Wireguard integration

### DISCUSSION THREAD

**Discussion by osevan** [netblue30/firejail#6488] [Ideas]
Dear community,

we could create wireguard Integration inside firejail networknameservice netns like with --net=br0 and all needed stuff in one way with additional --wireguard=/etc/wireguard/wg0.conf  pointing to config file ,for better integration inside firejail.

But we need wireguard function and --net=br0 functionality in same netns like wg0

What we expect is, we can connect every firejail sandbox together with wg0.conf standard config, and share resources , outside of server.

What you think about? 

Thanks and

Best Regards
 
I have here some examples for devs:
https://www.wireguard.com/netns/

And here some wg0.conf describing for easy Integration inside firejail netns with --wireguard=pathtoconfig
https://www.stavros.io/posts/how-to-configure-wireguard/

**Comment by ManuLinares** [comment]
Is there an update to this?

I can't 

    firejail --noprofile --net=myWireguardVpn firefox

because of

    Error: the software is not supported for /31 networks

---

Can't force 'firefox' to be only on my "myWireguardVpn" interface.

**Comment by lordkitsuna** [comment]
so i tested to verify, you can solve this by just giving the adapter a larger subnet. even a /30 seems to be enough. since most people run the wg net on its own subnet anyway this should be an easy solution. makes sense that it needs an address to give
